### PR TITLE
pubsub: switch to the new API

### PR DESCRIPTION
**Pull request by rakyll**
Related to https://github.com/GoogleCloudPlatform/google-cloud-go/commit/f4eafa8cbb48e8ef5e23b542c3cb51cd1cbce787.

### CODE REVIEW

File: pubsub/subscriptions/main.go [hunk 1]
```diff
@@ -89,12 +89,9 @@ func pullMsgs(client *pubsub.Client, name string, topic *pubsub.Topic) error {
 
 	// publish 10 messages on the topic.
 	for i := 0; i < 10; i++ {
-		_, err := topic.Publish(ctx, &pubsub.Message{
+		topic.Publish(ctx, &pubsub.Message{
```
Comment (rakyll) [inline pubsub/subscriptions/main.go]: @jba, what's our best practice here? Do we allow fire and forget?

Comment (broady) [inline pubsub/subscriptions/main.go]: Save the future result and check for errors after all of the messages are sent.

File: pubsub/topics/main.go [hunk 2]
```diff
@@ -111,15 +111,14 @@ func publish(client *pubsub.Client, topic, msg string) error {
 	ctx := context.Background()
 	// [START publish]
 	t := client.Topic(topic)
-	msgIDs, err := t.Publish(ctx, &pubsub.Message{
+	result := t.Publish(ctx, &pubsub.Message{
 		Data: []byte(msg),
 	})
+	id, err := result.Get(ctx)
```
Comment (broady) [inline pubsub/topics/main.go]: Add a comment above describing what this is doing.

Comment (rakyll) [inline pubsub/topics/main.go]: Done

File: pubsub/subscriptions/main.go (hunk 1 above)
Comment (rakyll) [inline pubsub/subscriptions/main.go]: Done, but it is awkward to do it in the same block. The developer would question why there is no blocking publish and what the merits of non-blocking publish has here. It is not quite a good experience having to write all the boilerplate to achieve a simple blocking publish.

File: pubsub/subscriptions/main.go [hunk 3]
```diff
@@ -13,13 +13,11 @@ import (
 	"time"
 
 	// [START imports]
-	"golang.org/x/net/context"
-
+	"cloud.google.com/go/iam"
 	"cloud.google.com/go/pubsub"
+	"golang.org/x/net/context"
 	"google.golang.org/api/iterator"
 	// [END imports]
```
Comment (broady) [inline pubsub/subscriptions/main.go]: Revert this re-ordering

File: pubsub/subscriptions/main.go [hunk 4]
```diff
@@ -88,12 +87,19 @@ func pullMsgs(client *pubsub.Client, name string, topic *pubsub.Topic) error {
...
 		})
+		results = append(results, res)
+	}
+
+	// Check if all messages are published.
+	for _, r := range results {
+		_, err := r.Get(ctx)
 		if err != nil {
-			return err
+			log.Printf("Publish error: %v", err)
```
Comment (broady) [inline pubsub/subscriptions/main.go]: This should return the error

**Comment by broady** [comment]
other samples need updating, too

```
Step #2: getting-started/bookshelf/app/app.go:301: assignment count mismatch: 2 = 1
Step #2: appengine_flexible/pubsub/pubsub.go:101: assignment count mismatch: 2 = 1
```

File: pubsub/subscriptions/main.go (hunk 4 above)
Comment (broady) [inline pubsub/subscriptions/main.go]: I did it (pushed to your branch)

**Comment by rakyll** [comment]
LGTM

**Pull request merged**